rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -195,6 +206,17 @@ func (p *BatchProposer) updateDBBatchInfo(batch *encoding.Batch, codecVersion en
 		p.recordAllBatchMetrics(metrics)
 	}
 
+	if len(batch.Chunks) > 0 && len(batch.Chunks[len(batch.Chunks)-1].Blocks) > 0 {
+		lastBlock := batch.Chunks[len(batch.Chunks)-1].Blocks[len(batch.Chunks[len(batch.Chunks)-1].Blocks)-1]
```

Comment:
too complicated expression.
```
lastChunk := batch.Chunks[len(batch.Chunks)-1]
lastBlock := lastChunk.Blocks[len(lastChunk.Blocks)-1]
```

Reply:
yeah. changed in [99916b2](https://github.com/scroll-tech/scroll/pull/1520/commits/99916b25fb58150d16e07dd8c002b9bd2d8c7457).